index.bs

```diff
@@ -656,10 +662,12 @@ To <dfn>process the lock request queue</dfn> |queue|:
         1. Let |mode| be |request|'s [=lock request/mode=].
         1. Let |callback| be |request|'s [=lock request/callback=].
         1. Let |p| be |request|'s [=lock request/promise=].
+        1. Let |signal| be |request|'s [=lock request/signal=].
         1. Let |waiting| be [=a new promise=].
         1. Let |lock| be a new [=lock-concept|lock=] with [=lock-concept/agent=] |agent|, [=lock-concept/clientId=] |clientId|, [=lock-concept/origin=] |origin|, [=lock-concept/mode=] |mode|, [=lock-concept/name=] |name|, [=lock-concept/released promise=] |p|, and [=lock-concept/waiting promise=] |waiting|.
         1. [=set/Append=] |lock| to |origin|'s [=held lock set=].
         1. [=Enqueue the following steps=] on |callback|'s [=relevant settings object=]'s [=responsible event loop=]:
+            1. If |signal| is present, then <a for=AbortSignal>remove</a> the algorithm [=signal to abort the request=] |request| from |signal|.
```

Comment:
Hmmm... Adding/removing should be on the event loop, as you've done here. But it seems like there's a potential race between aborting (e.g. calling abort(), which be within a task the event loop) and the lock being granted (on the lock task queue) that this doesn't solve. 

Before telling script that it was granted the lock, do we need to verify that it wasn't aborted and drop the lock if so?

Reply:
That sounds about right. I'll add that.